cpp/tests/sg/rf_test.cu

```diff
@@ -181,28 +187,85 @@ std::ostream& operator<<(std::ostream& os, const RfTestParams& ps)
 }
 
 template <typename DataT, typename LabelT>
-auto FilPredict(const raft::handle_t& handle,
-                RfTestParams params,
-                DataT* X_transpose,
-                RandomForestMetaData<DataT, LabelT>* forest)
+std::shared_ptr<thrust::device_vector<LabelT>> FilPredict(
+  const raft::handle_t& handle,
+  RfTestParams params,
+  DataT* X_transpose,
+  RandomForestMetaData<DataT, LabelT>* forest)
 {
-  auto pred = std::make_shared<thrust::device_vector<float>>(params.n_rows);
+  auto pred      = std::shared_ptr<thrust::device_vector<LabelT>>();
+  auto workspace = std::shared_ptr<thrust::device_vector<DataT>>();  // Scratch space
+  if constexpr (std::is_integral_v<LabelT>) {
+    // For classifiers, allocate extra scratch space to store probabilities from FIL
+    // We will perform argmax to convert probabilities into class outputs.
+    pred      = std::make_shared<thrust::device_vector<LabelT>>(params.n_rows);
+    workspace = std::make_shared<thrust::device_vector<DataT>>(params.n_rows * params.n_labels);
```

Comment:
Is `thrust::device_vector` memory use tracked by RMM?

Reply:
It won't be, but I suppose it is of less focus because this is in a test.